cellengine/resources/gate.py

```diff
@@ -1,155 +1,118 @@
-                to the server. If specified, do not specify ``fcs_file_id``.
-            locked (bool): Prevents modification of the gate via the web interface.
-            create_population (bool): Automatically create corresponding population.
-        """
-        if type(gates) is list:
-            return [cls._build_gate(gate) for gate in gates]
+def exception_handler(func):
+    def inner_function(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except Exception as err:
+            if type(err) is ValueError:
+                raise err
+            raise RuntimeError(f"Incorrect arguments passed to {func.__name__}.")
+
+    return inner_function
+
+
+def deep_update(source, overrides):
+    """
+    Update a nested dictionary or similar mapping.
+    Modify ``source`` in place.
+    """
+    for key, value in overrides.items():
+        if isinstance(value, Mapping) and value:
+            returned = deep_update(source.get(key, {}), value)
+            source[key] = returned
         else:
-            return cls._build_gate(gates)
+            source[key] = overrides[key]
+    return source
+
+
+@define(repr=False)
+class Gate:
+    _id: str = field(on_setattr=readonly)
+    experiment_id: str = field(on_setattr=readonly)
+    gid: str = field(on_setattr=readonly)
+    type: str
+    x_channel: str
+    model: Dict
+    fcs_file_id: Optional[str] = None
+    tailored_per_file: Optional[bool] = None
+    names: Optional[List[str]] = field(default=None)
+    name: Optional[str] = field(default=None)
+    y_channel: Optional[str] = field(default=None)
+
+    def __repr__(self):
+        if self.name:
+            label, name = ("name", self.name)
+        else:
+            label, name = ("names", str(self.names))
+        return f"{self.type}(_id='{self._id}', {label}='{name}')"
 
-    @classmethod
-    def _build_gate(cls, gate):
-        """Get the gate type and return instance of the correct subclass."""
-        module = importlib.import_module(__name__)
-        gate, _ = cls._separate_gate_and_population(gate)
-        gate_type = getattr(module, gate["type"])
-        return gate_type(properties=gate)
+    @staticmethod
+    def _format_gate(gate, **kwargs):
+        module = importlib.import_module("cellengine")
+        _class = getattr(module, gate["type"])
+        return _class._format(**gate)
 
     @classmethod
-    def _separate_gate_and_population(cls, gate):
-        try:
-            if "gate" in gate.keys():
-                return gate["gate"], [k for k in gate.keys() if k != "gate"]
-            else:
-                return gate, None
-        except KeyError:
-            raise ValueError("Gate payload format is invalid")
+    def create_many(
+        cls,
+        gates: List[Dict],
+    ):
+        experiment_id = set([g["experiment_id"] for g in gates])
+        if len(experiment_id) == 1:
+            experiment_id = experiment_id.pop()
+        else:
```

Comment:
Looks like `experiment_id` is unused after this, so could be
```suggestion
        if len(experiment_id) != 1:
```

Reply:
:+1: